test/pool-close.ts

```diff
@@ -0,0 +1,70 @@
+import { test } from 'tap';
+import Piscina from '..';
+import { resolve } from 'path';
+
+test('close()', async (t) => {
+  t.test('no pending tasks', async (t) => {
+    const pool = new Piscina({ filename: resolve(__dirname, 'fixtures/eval.js') });
+    await pool.close();
+    t.pass('pool closed successfully');
+  });
+
+  t.test('queued tasks waits for all tasks to complete', async (t) => {
+    const pool = new Piscina({ filename: resolve(__dirname, 'fixtures/eval.js'), maxThreads: 1 });
+
+    const task1 = pool.run('1 + 1');
+    const task2 = pool.run('2 + 2');
```

Comment:
I'd recommend using more complex tasks, like a `setTimeout` with more than a few milliseconds to test out it truly flushes the queue within a time window.